lib/coverband/at_exit.rb

```diff
@@ -5,14 +5,30 @@ class AtExit
     @semaphore = Mutex.new
 
     def self.register
+      return if ENV['COVERBAND_DISABLE_AT_EXIT']
       return if @at_exit_registered
+
       @semaphore.synchronize do
         return if @at_exit_registered
+
         @at_exit_registered = true
         at_exit do
           ::Coverband::Background.stop
-          Coverband.report_coverage(true)
-          Coverband.configuration.logger&.debug('Coverband: Reported coverage before exit')
+
+          #####
+          # TODO: This is is brittle and not a great solution to avoid deploy time
+          # actions polluting the 'runtime' metrics
+          #
+          # * should we skip /bin/rails webpacker:compile ?
+          # * Perhaps detect heroku deployment ENV var opposed to tasks?
+          #####
+          default_heroku_tasks = ['assets:clean', 'assets:precompile']
+          if defined?(Rake) && Rake.application.top_level_tasks.any? { |task| default_heroku_tasks.include?(task) }
```

Comment:
Getting this error when I exit from puma on coverband_demo:
```
/Users/karlbaum/workspace/coverband/lib/coverband/at_exit.rb:26:in `block (2 levels) in register': undefined method `application' for Rake:Module (NoMethodError)
```

Reply:
cool fixed, will push soon